### ISSUE TITLE
Consider some adminy API calls as within scope?

### ISSUE THREAD

**Issue by denuno** [llorllale/youtrack-api#238]
I've looked at this implementation a couple of times, and while I'd love to replace the little Groovy API facade I wrote with this, I can't without some of the stuff needed to run my unit tests, which include creating a project, creating custom fields, and custom field value bundles (States, Versions, and the aforementioned custom field ENUM bundles).  It handles adding users and whatnot too but that's not as important as being able to set up a project for testing.

Maybe I should just release what I've done as YAYouAF (Yet Another YouTrack API Facade)... it uses Gradle for its build and can automatically set up a YouTrack docker image for tests (on *nix), so it's got that going for it, and it covers the whole API I think (needed since it takes a vanilla YT docker image and "Makes it so", if required, before testing-- also handy for re-populating a YT instance, where DBs can grow astonishingly fast!)... but meh.

I'm at ~800 lines in a single fluentish Groovy file, and it uses ReST everywhere except the 2 places XML is required (as JB said somewhere that someday they'll switch to JSON), but I'd ditch it in favor of a Somebody Else's Problem-- or even the potentiality of such.

With that name I just came up with alone I'm starting to lean towards adding another Java facade to the fray, but maybe I overlooked something-- is adding a Version to a version bundle considered within the scope this project's API?  I didn't see anything for it but if I'm mistaken, heck, that'd probably be enough to switch right there.  ('Tis the only hard test requirement for the non-setup-ish stuff, as it is tied to our build/release process)

Long story short: Any plans on supporting API calls besides the non-admin per se ones?

**Comment by llorllale** [comment]
@denuno wall of text :O

So it sounds like you'd like a nice facade that provides facilities for creating/configuring projects, adding users, etc., in order to setup your test environment for some other project. Correct me if I'm wrong.

> Any plans on supporting API calls besides the non-admin per se ones?

I think my decision for this to be an API focused on a *user* (ie. a DEV) can be changed. 

I originally built this library because I wanted to develop an IDE plugin. That never went anywhere (and YouTrack is not used in my current work environment, sadly).

So the answer is **I am willing to extend the scope to include admin stuff**.

If that's ok with you then we can start by opening new issues focused on specific features that you want. :)

**Comment by llorllale** [comment]
@denuno have I answered your question?

**Comment by denuno** [comment]
@llorllale I'm wont to write-- that was a short one even! :smiley: 

You're exactly right.  I'll open an issue for support for adding Bundles first.  Do you think one issue for bundles in general or issues for each bundle type?  VersionBundles would be my first choice, FWIW.

Thanks for being willing to extend the scope!

**Comment by llorllale** [comment]
@denuno please open for VersionBundles first
